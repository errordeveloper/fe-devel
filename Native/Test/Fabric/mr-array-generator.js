FC = require("Fabric").createClient();

cv = FC.MR.createConstValue("Size", 10);

ago = FC.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(Size index, io Scalar output) { output = Scalar(index); }", "foo");
ag = FC.MR.createArrayGenerator(cv, ago);

mo = FC.KLC.createMapOperator("bar.kl", "operator bar(Scalar input, io Scalar output) { output = input * input; }", "bar");
m = FC.MR.createMap(ag, mo);

var count = m.getCount();
console.log("count = " + count);
for (var i=0; i<count; ++i)
  console.log("produce("+i+") = "+m.produce(i));

FC.close();
