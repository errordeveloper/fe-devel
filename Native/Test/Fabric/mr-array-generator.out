count = 10
produce(0) = 0
produce(1) = 1
produce(2) = 4
produce(3) = 9
produce(4) = 16
produce(5) = 25
produce(6) = 36
produce(7) = 49
produce(8) = 64
produce(9) = 81
